core/src/main/java/org/opensearch/sql/expression/function/BuiltinFunctionRepository.java

```diff
@@ -182,6 +182,13 @@ private FunctionBuilder getFunctionBuilder(
     if (isCastFunction(functionName) || sourceTypes.equals(targetTypes)) {
       return funcBuilder;
     }
+    // For functions with variable number of args (ex: concat())
+    // targetTypes will always be empty (as the function signature is not fixed),
```

Comment:
Could you elaborate why `targetTypes` will always be empty even if signature for `concat` is already found?

Reply:
`targetTypes` will always be empty, because the functions' definitions (`FunctionDSL.java` code) run (signatures get resolved) during the server start, not runtime. So, we can't have a precisely defined signature for a variable arguments function, as it expects array, which on startup is defined as empty. Hence, during runtime, when we call the concat() function with different number of args, it will never match the defined function signature with empty array.

Reply:
If that's the case, does it make sense just to return: 
```
return funcBuilder;
```
Does `castArguments(sourceTypes, sourceTypes, funcBuilder)` actually do anything for us?  What if sourceTypes is null or unknown?

Reply:
Addressed in bed81bfefc360c00ed64e9609ceb3f02f2513fc9